client/src/app/users/user.service.spec.ts

```diff
@@ -297,15 +297,17 @@ describe('UserService', () => {
 
   describe('Adding a user using `addUser()`', () => {
     it('talks to the right endpoint and is called once', waitForAsync(() => {
+      const user_id = 'pat_id';
```

Comment:
The previous test here didn't work because we were expecting to get a `string` ID back, when the HTTP request actually returns a _JSON object_ back, which contains an `id` field that has the `string` ID.

This is arguably a type error, and one might hope that the TypeScript type checker would have helped us here. Unfortunately the Mockito `.returnValue()` method can take any value so the type system couldn't tell that we were passing a `string` when a JSON object was expected.